File: client-domain/src/main/java/com/jraska/github/client/Urls.java
package com.jraska.github.client;

import com.jraska.github.client.common.Preconditions;

import okhttp3.HttpUrl;

public final class Urls {
  public static HttpUrl user(String login) {
    return HttpUrl.parse("https://github.com/" + login);
  }

  public static HttpUrl repo(String fullPath) {
    return HttpUrl.parse("https://github.com/" + fullPath);
  }

  private Urls() {
    Preconditions.throwNoInstances();
  }
}

File: client-domain/src/main/java/com/jraska/github/client/Urls.kt
package com.jraska.github.client

import okhttp3.HttpUrl

class Urls private constructor() {
  companion object {
    fun user(login: String): HttpUrl {
      return HttpUrl.parse("https://github.com/" + login) as HttpUrl
    }

    fun repo(fullPath: String): HttpUrl {
      return HttpUrl.parse("https://github.com/" + fullPath) as HttpUrl
    }
  }
}
